### ISSUE TITLE
IDE / Editor

### ISSUE THREAD

**Issue by GWRon** [bmx-ng/bmx-ng#26]
Asking whether one of you has a working - or started working on a - editor which is a bit more advanced than MaxIDE ?
I know Brucey had a peek at IntelliJ but I think there is nothing "tryable" yet. 

Are there people out there interested in building up a plugin for one of the major/public IDEs? Means Atom, IntelliJ, Visual Studio Code...

Doing it "in public" allows for others to contribute or at least do discuss things if there is something to clarify.

**Comment by Kerntrick** [comment]
I use Blide with NG. it's not perfect but works good enough. You need a custom bmk file for some things as blide doesn't expose it's call to bmk.  
I think the easiest path forward for a new Ide would "Visual studio code".  but I've tired in the past to find some information about how to write a custom parser and there is not much out there.

**Comment by GWRon** [comment]
Thanks for joining me in the discussion @Kerntrick. I was thinking about cross platform IDEs, so BLIde is no option (I know you did not suggest it).

I wonder why you suggest Visual Studio Code if you already tried to look for tutorials?
I once in a while am looking for tutorials - and eg. for IntellJ there are plenty of them.

**Comment by Kerntrick** [comment]
I just did a basic google search for any tutorials about writing a custom parser the ide. but there are a lot languages already supported with good levels of intellisense.  I didn't dive very deep into it, but it may be a issue where the best way to do it is to look at some already implemented parsers and see how they work.  
I agree that to handle large projects in NG needs a good ide with intellisence.  I only suggested VScode because it seems custom plugins pop up fairly quickly for languages and I've never used intellij so I can't comment on it.

**Comment by HenriVi** [comment]
Hi,

I was thinking about writing an IDE for NG. I like MaxIDE, but lack of autocomplete is a no-go nowadays. I currently use wxMax for work, so probably want to use it. Benefits are crossplatform, and lot of things already available for it (like Bmax lexer). I want to keep everything as simple and easy as possible, with emphasis to reliability.

**Comment by woollybah** [comment]
Yes, I've started some work on an Intellij language plugin. Nothing to show for it yet though.
Intellij has some very good language facilities available, which in theory would make things like intellisense less of a chore to integrate.
But since I have a million things to work on regarding BlitzMax, it isn't high on my priority list at the moment.

**Comment by HenriVi** [comment]
No worries there. It is important that someone focuses on the language itself.

**Comment by HenriVi** [comment]
Okay,

I have setup a Github repo for this 'Editor'-project.

You can follow progress here https://github.com/HenriVi/Editor
